File: task-tracker-ui/src/app/model/space-user/GetSpaceUserRequest.ts
import PagedRequest, { mapPagedRequest } from "../PagedRequest";

export default interface GetSpaceUserRequest extends PagedRequest {
    userId?: number,
    spaceId?: number,
    skipUserIds?: number[]
}

export const mapGetSpaceUserRequest = (request: GetSpaceUserRequest) => {
    let params = mapPagedRequest(request)

    if(request.userId) {
        params = params.append('UserId', request.userId)
    }

    if(request.spaceId) {
        params = params.append('SpaceId', request.spaceId)
    }
    if(request.skipUserIds) {
        request.skipUserIds.forEach(id => params = params.append('SkipUserIds', id))
    }

    return params
}

File: TaskTracker.Model/SpaceUser/Request/GetSpaceUserRequest.cs
using TaskTracker.Model.Attribute;
using TaskTracker.Model.Enum;

namespace TaskTracker.Model.SpaceUser.Request
{
    public class GetSpaceUserRequest : PagedRequest
    {
        public long? UserId { get; set; }
        public long? SpaceId { get; set; }
        [CustomFilter(ComparisonType = ComparisonType.DoesNotContain, PropertyName = "UserId")]
        public ICollection<long>? SkipUserIds { get; set; }
    }
}
